### ISSUE TITLE
Support custom private chains that are not extending any known public chains

### ISSUE THREAD

**Issue by NicolasMassart** [ethereumjs/ethereumjs-monorepo#769]
Using a custom private chain requires to use ethereumjs-common:
```
    const Common = require('ethereumjs-common').default
    const customCommon = Common.forCustomChain(
      'mainnet',
      {
        name: 'custom_ibft2_chain',
        networkId: 1664,
        chainId: 1664,
      },
      'petersburg',
    )
```
But it's required to use chain and hardfork params that means nothing here as the custom private chain doesn't inherit from mainnet or any known public testnets and the hardfork is not relevant either. So I still have to use valid "mainnet" and "petersburg" values to have my transaction work, but it means nothing in my context.
I think that would be nice if these params could be optional.
Let me know if there's another way to build a custom common config.
Thanks.

**Comment by alcuadrado** [comment]
Hey @NicolasMassart,

That's just a helper method that I introduced for convenience. You can use the constructor and pass the chain params you want.

You would have to use the standard HF names though, as the different ethereumjs libs assume those names.

**Comment by NicolasMassart** [comment]
Hum, so should I be using the latest fork name by default in my case?
Would you have some link to a doc about creating this object without the constructor or do I have to read the TS code?

**Comment by alcuadrado** [comment]
I'm afraid that's not documented nor has proper TS types. That's why the utility method is so handy.

I initialize a custom `Common` [here](https://github.com/nomiclabs/buidler/blob/development/packages/buidler-core/src/internal/buidler-evm/provider/node.ts#L186). It may be helpful to read that code.

**Comment by holgerd77** [comment]
Solved by #1034 (not yet released but feel free to request a prioritized release on this), will close.
